datafusion/pruning/src/pruning_predicate.rs

```diff
@@ -1105,6 +1106,20 @@ fn rewrite_expr_to_prunable(
             None,
         ));
         Ok((left, op, right))
+    } else if let Some(cast_col) = column_expr_any.downcast_ref::<CastColumnExpr>() {
```

Comment:
@kosiew  I can't remember -- did we ever find a way to unify the CastColumnExpr and CastExpr? This is a good example where the code is almost identical for both and the duplication seems unfortunate

 I think it is tracked via 
- https://github.com/apache/datafusion/issues/17765

This seems to have come in via
- #17773